acceptance/analyzer_test.go

```diff
@@ -0,0 +1,503 @@
+)
+
+var (
+	analyzeDockerContext = filepath.Join("testdata", "analyzer", "analyze-image")
+	analyzerBinaryDir    = filepath.Join("testdata", "analyzer", "analyze-image", "container", "cnb", "lifecycle")
+	analyzeImage         = "lifecycle/acceptance/analyzer"
+	analyzerPath         = "/cnb/lifecycle/analyzer"
+	cacheImageRepo       = "lifecycle/acceptance/analyzer/cache-image"
+)
+
+var (
+	registry *ih.DockerRegistry
+)
+
+func TestAnalyzer(t *testing.T) {
+	dockerConfigDir, err := ioutil.TempDir("", "test.docker.config.dir")
+	h.AssertNil(t, err)
+	defer os.RemoveAll(dockerConfigDir)
+
+	registry = ih.NewDockerRegistryWithAuth(dockerConfigDir)
+	registry.Start(t)
+	defer registry.Stop(t)
+
+	os.Setenv("DOCKER_CONFIG", registry.DockerDirectory)
+
+	outDir, err := filepath.Abs(filepath.Join("..", "out"))
+	h.AssertNil(t, err)
+
+	info, err := h.DockerCli(t).Info(context.TODO())
+	h.AssertNil(t, err)
+	osType := info.OSType
+
+	h.ConditionallyBuildLifecycle(t, outDir, osType)
+	h.CopyLifecycle(t, filepath.Join(outDir, osType, "lifecycle"), analyzerBinaryDir)
+
+	buildTestImage(t, analyzeImage, analyzeDockerContext)
+	defer removeTestImage(t, analyzeImage)
+	spec.Run(t, "acceptance", testAnalyzer, spec.Parallel(), spec.Report(report.Terminal{}))
+}
+
+func testAnalyzer(t *testing.T, when spec.G, it spec.S) {
+	when("called without an app image", func() {
+		it("errors", func() {
+			cmd := exec.Command(
+				"docker", "run", "--rm",
+				"--env", "CNB_REGISTRY_AUTH={}",
+				analyzeImage,
+				analyzerPath,
+			)
+			output, err := cmd.CombinedOutput()
+
+			h.AssertNotNil(t, err)
+			expected := "failed to parse arguments: received 0 arguments, but expected 1"
+			h.AssertStringContains(t, string(output), expected)
+		})
+	})
+
+	when("cache image tag and cache directory are both blank", func() {
+		it("warns", func() {
+			cmd := exec.Command(
+				"docker", "run", "--rm",
+				"--env", "CNB_REGISTRY_AUTH={}",
+				analyzeImage,
+				analyzerPath,
+				"some-image",
+			)
+			output := h.Run(t, cmd)
+
+			expected := "Not restoring cached layer metadata, no cache flag specified."
+			h.AssertStringContains(t, output, expected)
+		})
+	})
+
+	when("the provided layers directory isn't writeable", func() {
+		it("recursively chowns the directory", func() {
+			cmd := exec.Command(
+				"docker", "run", "--rm",
+				"--env", "CNB_REGISTRY_AUTH={}",
+				analyzeImage,
+				"/bin/bash", "-c",
+				fmt.Sprintf("%s some-image; ls -al /layers", analyzerPath),
+			)
+			output := h.Run(t, cmd)
+
+			h.AssertMatch(t, output, "2222 3333 .+ \\.")
+			h.AssertMatch(t, output, "2222 3333 .+ group.toml")
+		})
+	})
+
+	when("group path is provided", func() {
+		it("uses the provided group path", func() {
+			cmd := exec.Command(
+				"docker", "run", "--rm",
+				"--env", "CNB_REGISTRY_AUTH={}",
+				analyzeImage,
+				analyzerPath,
+				"-group", "/layers/other-group.toml",
+				"some-image",
+			)
+			h.Run(t, cmd)
+		})
+	})
+
+	when("daemon case", func() {
```

Comment:
```suggestion
	when("mounted daemon", func() {
```

Reply:
@dfreilich I like "daemon case" - there is slight precedent for it elsewhere in the code and RFCs (see [here](https://github.com/buildpacks/lifecycle/blob/1ef63ec5114e90962c0a4ee2dc26a278e910d84d/metadata.go#L87) and [here](https://github.com/buildpacks/rfcs/blob/main/text/0026-lifecycle-all.md)). Do you feel strongly about it?

Reply:
I am going to keep "daemon case" for now.